languages/go/exercises/concept/numbers-floating-point/.docs/instructions.md

```diff
@@ -0,0 +1,41 @@
+In this exercise you'll be working with savings accounts. Each year, the balance of your savings account is updated based on its interest rate. The interest rate your bank gives you depends on the amount of money in your account (its balance):
+
+- -3.213% for a negative balance.
+- 0.5% for a positive balance less than `1000` dollars.
+- 1.621% for a positive balance greater or equal than `1000` dollars and less than `5000` dollars.
+- 2.475% for a positive balance greater or equal than `5000` dollars.
+
+You have three tasks, each of which will deal your balance and its interest rate.
+
+### 1. Calculate the interest rate
+
+Implement the `InterestRate` method to calculate the interest rate based on the specified balance:
+
+```golang
+InterestRate(200.75)
+// 0.5
+```
+
+Note that the value returned is a `float64`.
+
+### 2. Calculate the annual balance update
+
+Implement the `AnnualBalanceUpdate` method to calculate the annual balance update, taking into account the interest rate:
+
+```golang
+AnnualBalanceUpdate(200.75)
```

Comment:
Hi, yes I agree but I think I completely miss the point: I thought I need to copy the exact exercise transposing to golang. Can we change completely instead?

Reply:
We have 2 options here I think:
1) deviate from the original C# exercise in the story we tell here so it is not about currency
2) tell the student that normally we wouldn't use floats for representing currency but here we will still do it for the sake of this exercise

I would vote for option 1) and it looks to me we all agree on that.

@imanc: can you think of another storyline here?

Reply:
What were the C# people thinking in the first place? 🤔

Reply:
doesn't C# have a decimal type?

Reply:
> What were the C# people thinking in the first place?

Good question :) In C#, there are three floating-point types: `double`/`float`/`decimal`. The latter is exclusively used in financial code, and is the recommend type for doing so. One should never use `double`/`float` for storing money though, but that is also explained in the exercise. In short, while the exercise might work well for C#/F#, I can totally see why it wouldn't work here. As a consequence, it makese sense to adapt/rewrite the theme to have something that works for Go. Note that this can and will happen when porting exercises, as languages will implement concepts differently.